src/amplify.ts

```diff
@@ -0,0 +1,155 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import * as cloudflare from "@pulumi/cloudflare";
+import * as std from "@pulumi/std";
+
+type CustomRule = {
+  source: string;
+  status: string;
+  target: string;
+};
+
+export interface AmplifyAppConfig {
+  name: string;
+  githubUrl: string;
+  githubAccessToken?: string;
+  domainName: string;
+  cloudflareDomain?: string;
+  cloudflareZoneId?: string;
+  prefix: string;
+  branchName: string;
+  environmentVariables: Record<string, string | pulumi.Output<any>>;
+  tags: { [key: string]: string };
+  enableBasicAuth?: boolean;
+  basicAuthUsername?: string;
+  basicAuthPassword?: string;
+  platform?: "WEB" | "WEB_COMPUTE";
+  buildCommand: string;
+  preBuildCommand?: string;
+  artifactsBaseDirectory: string;
+  monorepoAppRoot?: string;
+  nodeVersion?: string;
+  customRules?: CustomRule[];
+}
+
+export function createAmplifyApp(config: AmplifyAppConfig) {
+  const name = `${config.prefix}.${config.domainName}`;
+
+const buildSpec = `version: 1
+applications:
+  - frontend:
+      phases:
+        preBuild:
+          commands:
+            - ${config.preBuildCommand || ''}
+        build:
+          commands:
+            - ${config.buildCommand}
+      artifacts:
+        baseDirectory: ${config.artifactsBaseDirectory}
+        files:
+          - '**/*'
+      cache:
+        paths:
+          - node_modules/**/*
+    appRoot: ${config.monorepoAppRoot || 'app'}
+`;
+
+const amplifyApp = new aws.amplify.App(name, {
+    name: name,
+    repository: config.githubUrl,
+    oauthToken: config.githubAccessToken,
+    platform: config.platform || "WEB_COMPUTE",
+    buildSpec: buildSpec,
+    customRules: config.customRules || [
+      {
+        source: "/<*>",
+        status: "404",
+        target: "/index.html",
+      },
+    ],
+    environmentVariables: {
+      AMPLIFY_DIFF_DEPLOY: "false",
+      AMPLIFY_MONOREPO_APP_ROOT: config.monorepoAppRoot || "app",
+      ...config.environmentVariables,
+    },
+    enableBasicAuth: config.enableBasicAuth || false,
```

Comment:
I think we can remove this for now. We have no use-case for basic auth & will drop 3 parameter (easier to configure it ).
But it's ok for me if we leave it as it is now